browser/js/common/directives/cart/cart.js

```diff
@@ -1,8 +1,65 @@
 'use strict';
 
-app.directive('cart', function() {
-  return{
-		restrict: 'E',
-		templateUrl:'/js/common/directives/cart/cart.html'
-	}
-})
+app.directive('cart', function (CartFactory) {
+    return {
+        restrict: 'E',
+        templateUrl: '/js/common/directives/cart/cart.html',
+        link: function (scope) {
+            CartFactory.getItems()
+                .then(function (products) {
+                    // scope.products = products;
+                    var count = products.reduce(function (a, b) {
+                        return a + b.quantity
+                    }, 0)
+                    CartFactory.count = count
+                    scope.count = CartFactory.count
+                })
+        }
+    };
+});
+
+app.config(function ($stateProvider) {
+    $stateProvider.state('inCart', {
+        url: "/cart",
+        controller: "CartController",
+        templateUrl: "/js/common/directives/cart/incart.html",
+        resolve: {
+            products: function (CartFactory) {
+                return CartFactory.getItems()
+                    .then(function (list) {
+                        return list;
+                    });
+
+            }
+        }
+    });
+});
+
+app.controller('CartController', function ($scope, products, CartFactory, $state) {
+    $scope.nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    $scope.toUpdate = null;
+    $scope.userId = CartFactory.userId;
+    $scope.products = products;
+
+    $scope.editItem = CartFactory.editItem
+
+    $scope.deleteItem = function (product) {
+        CartFactory.deleteItem(product)
```

Comment:
The put should force CartFactory to update. This functionality shouldn't be here